src/algebra/ordered_group.lean

```diff
@@ -33,400 +57,447 @@ with a partial order in which multiplication is strictly monotone. -/
 @[protect_proj, ancestor comm_group partial_order]
 class ordered_comm_group (α : Type u) extends comm_group α, partial_order α :=
 (mul_le_mul_left : ∀ a b : α, a ≤ b → ∀ c : α, c * a ≤ c * b)
-
 attribute [to_additive] ordered_comm_group
 
 @[to_additive]
-instance units.covariant_class [ordered_comm_monoid α] :
-  covariant_class (units α) (units α) (*) (≤) :=
-{ covc := λ a b c bc, by {
-  rcases le_iff_eq_or_lt.mp bc with ⟨rfl, h⟩,
-  { exact rfl.le },
-  refine le_iff_eq_or_lt.mpr (or.inr _),
-  refine units.coe_lt_coe.mp _,
-  cases lt_iff_le_and_ne.mp (units.coe_lt_coe.mpr h) with lef rig,
-  exact lt_of_le_of_ne (mul_le_mul_left' lef ↑a) (λ hg, rig ((units.mul_right_inj a).mp hg)) } }
+instance ordered_comm_group.to_covariant_class_left_le (α : Type u) [ordered_comm_group α] :
+  covariant_class α α (*) (≤) :=
+{ covc := λ a b c bc, ordered_comm_group.mul_le_mul_left b c bc a }
 
 /--The units of an ordered commutative monoid form an ordered commutative group. -/
 @[to_additive]
 instance units.ordered_comm_group [ordered_comm_monoid α] : ordered_comm_group (units α) :=
-{ mul_le_mul_left := λ a b h c, mul_le_mul_left' h _,
+{ mul_le_mul_left := λ a b h c, (mul_le_mul_left' (h : (a : α) ≤ (b : α)) _ :  (c : α) * a ≤ c * b),
   .. units.partial_order,
   .. (infer_instance : comm_group (units α)) }
 
-section ordered_comm_group
-variables [ordered_comm_group α] {a b c d : α}
-
-@[to_additive ordered_add_comm_group.add_lt_add_left]
-lemma ordered_comm_group.mul_lt_mul_left' (a b : α) (h : a < b) (c : α) : c * a < c * b :=
-begin
-  rw lt_iff_le_not_le at h ⊢,
-  split,
-  { apply ordered_comm_group.mul_le_mul_left _ _ h.1 },
-  { intro w,
-    replace w : c⁻¹ * (c * b) ≤ c⁻¹ * (c * a) := ordered_comm_group.mul_le_mul_left _ _ w _,
-    simp only [mul_one, mul_comm, mul_left_inv, mul_left_comm] at w,
-    exact h.2 w },
-end
-
-@[to_additive ordered_add_comm_group.le_of_add_le_add_left]
-lemma ordered_comm_group.le_of_mul_le_mul_left (h : a * b ≤ a * c) : b ≤ c :=
-have a⁻¹ * (a * b) ≤ a⁻¹ * (a * c), from ordered_comm_group.mul_le_mul_left _ _ h _,
-begin simp [inv_mul_cancel_left] at this, assumption end
-
-@[to_additive]
-lemma ordered_comm_group.lt_of_mul_lt_mul_left (h : a * b < a * c) : b < c :=
-have a⁻¹ * (a * b) < a⁻¹ * (a * c), from ordered_comm_group.mul_lt_mul_left' _ _ h _,
-begin simp [inv_mul_cancel_left] at this, assumption end
-
 @[priority 100, to_additive]    -- see Note [lower instance priority]
 instance ordered_comm_group.to_ordered_cancel_comm_monoid (α : Type u)
   [s : ordered_comm_group α] :
   ordered_cancel_comm_monoid α :=
 { mul_left_cancel       := @mul_left_cancel α _,
-  le_of_mul_le_mul_left := @ordered_comm_group.le_of_mul_le_mul_left α _,
+  le_of_mul_le_mul_left := λ a b c, (mul_le_mul_iff_left a).mp,
   ..s }
 
+/-  DT: Why does `by apply_instance` not work here? -/
```

Comment:
I will make a PR fixing this

Reply:
#7835

Reply:
After merging your PR into this one, `by apply_instance` works!  I therefore removed this instance, thanks!